Add GraphQL.SystemTextJson (#1512)
graphql-dotnet/graphql-dotnet: README.md

> If errors occur, it is recommended that you first check the behavior on the latest available preview version before
> reporting a issue.

> You can install the latest version via NuGet.

Reviewer: ```suggestion
You can install the latest stable version via [NuGet](https://www.nuget.org/packages/GraphQL).
```

Author: I'd removed this link as the GraphQL package itself is not really a valid entry point for most folks now. I'm basically now saying, "You can install from NuGet" then "Either one of these". The stable word is useful though.

Reviewer: I understand what it is about. But in fact, I always had in mind the idea of taking one more step forward and putting `IDocumentWriter` (and in future some other interfaces) in a separate package (GraphQL.Abstractions maybe) and make both serializers packages dependent on this package. The goal is to separate common abstractions from graphql engine implementation stuff. Ideally, this package should not be changed (often) and may be even located in a separate repository. So `GraphQL` package continues to act as _entry point_ and the user will install into app both packages - GraphQL + GraphQL.SomeSerializer (both dependant from GraphQL.Abstractions). @joemcbride @benmccallum How do you like that?

Reviewer: And I approve of your change pointing the user to myget instead of nuget since often people don’t know where to get the latest versions. The 3.0.0 version of the package is born for a very long time and intermediate previews of the version appear on the nuget with a delay (manually published). That is why I focused on the word **stable**, although there is also a bunch of preview versions. So it’s better to immediately send the reader to myget for latest versions as you point later in readme.

Author: What if we change the wording to something like:

> You can install the latest stable version via [NuGet](...).
> `> dotnet add package GraphQL`
> 
> For serialized outputs, you'll need an `IDocumentWriter` implementation. We support [GraphQL.SystemTextJson](...) for .NET Core 3+ else there's a legacy [GraphQL.NewtonsoftJson](...) (formerly included in [GraphQL](...),) or you can bring your own.
> `> dotnet add package GraphQL.SystemTextJson`
> `> dotnet add package GraphQL.NewtonsoftJson`
> 
> You can get the latest pre-release packages from the [MyGet feed](...), where you may want to explicitly pull a certain version using `-v`.
> `> dotnet add package GraphQL.SystemTextJson -v 3.0.0-preview-1448`

Reviewer: ok

Author: Tweaked it again slightly. I thought about it and the async write is only an issue in an aspnet context, so just put in a warning for that situation.

> If you're targeting .NET Core 3+, you should use [GraphQL.SystemTextJson](https://www.nuget.org/packages/GraphQL.SystemTextJson/).

Reviewer: ```suggestion
If you're targeting .NET Core 3+, you are advised to use [GraphQL.SystemTextJson](https://www.nuget.org/packages/GraphQL.SystemTextJson/).
```

Author: I'd say there's no point even suggesting otherwise, and it's highly risky like the guys from MS have said. I'd prefer to not even suggest it.

Reviewer: As you wish

```
> dotnet add package GraphQL.SystemTextJson
```

> Else, you should use the legacy [GraphQL.NewtonsoftJson](https://www.nuget.org/packages/GraphQL.NewtonsoftJson/)

Reviewer: ```suggestion
Else, you should use the legacy and time-tested [GraphQL.NewtonsoftJson](https://www.nuget.org/packages/GraphQL.NewtonsoftJson/)
```

Author: Similar to above, I don't think we should be promoting this at all.

Reviewer: ok

> reporting a issue. Latest 3.0.0-preview-XXXX versions are **backwards incompatible** with latest stable 2.4.0 version.
> You can see the changes in public APIs using [fuget.org](https://www.fuget.org/packages/GraphQL/3.0.0-preview-1352/lib/netstandard2.0/diff/2.4.0/).

You can install the latest stable version via [NuGet](https://www.nuget.org/packages/GraphQL/).
```
> dotnet add package GraphQL
```

For serialized results, you'll need an `IDocumentWriter` implementation.
We support [GraphQL.SystemTextJson](https://www.nuget.org/packages/GraphQL.SystemTextJson/) for .NET Core 3+,
[GraphQL.NewtonsoftJson](https://www.nuget.org/packages/GraphQL.NewtonsoftJson/)
(formerly included in [GraphQL](https://www.nuget.org/packages/GraphQL/)), or you can bring your own.
```
> dotnet add package GraphQL.SystemTextJson
> dotnet add package GraphQL.NewtonsoftJson
```
> *Note: You can use `GraphQL.NewtonsoftJson` with .NET Core 3+, just be aware it lacks async writing 
> capabilities so writing to an ASP.NET Core 3.0 `HttpResponse.Body` will require you to set 

> > `AllowSynchronousIO` to `false` as per [this announcement](https://github.com/aspnet/Announcements/issues/342);

Reviewer: ```suggestion
> `AllowSynchronousIO` to `true` as per [this announcement](https://github.com/aspnet/Announcements/issues/342);
```

Author: 🤦‍♂
